gamemode/obj_player_extend.lua

```diff
@@ -55,6 +55,75 @@ end
 
 net.Receive('regen_mana', function(len, ply)
   local mana = net.ReadUInt(9)
-  print('regen_mana')
   ply:SetMana(mana)
-end)
\ No newline at end of file
+end)
+
+function meta:GetStatus(type)
+  local ent = self["status_"..type]
+  if ent and ent:GetOwner() == self then 
+    print(ent:GetClass())
+    return ent end
+end
+
+function meta:GiveStatus(type, dur, Effector)
+  local alreadyexists = self:GetStatus(type)
+  if not alreadyexists then
+
+  local ent = ents.Create("status_"..type)
+  if ent:IsValid() then
+    ent:Spawn()
+    ent:SetPlayer(self)
+    if Effector then
+      ent:setEffector(Effector)
+    end
+    end
+  end
+end
+  local ent = ents.Create("status_" .. type)
```

Comment:
I think this might be duplicated from the previous lines 

```lua
  if ent:IsValid() then
    ent:Spawn()
    ent:SetPlayer(self)
    if Effector then
      ent:setEffector(Effector)
    end
    end
```